src/EPiServer.SocialAlloy.Web/Social/Tutorials/LikeButton/LikeButtonBlockController.cs

```diff
@@ -0,0 +1,150 @@
+        private readonly IContentRepository contentRepository;
+        private const int LIKED_RATING = 1;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        public LikeButtonBlockController()
+        {
+            // This is all wired up by the installation of the EPiServer.Social.Ratings.Site package
+            this.ratingService = ServiceLocator.Current.GetInstance<IRatingService>();
+
+            // This is wired up by EPiServer Core/Framework
+            this.contentRepository = ServiceLocator.Current.GetInstance<IContentRepository>();
+            this.pageRouteHelper = ServiceLocator.Current.GetInstance<IPageRouteHelper>();
+        }
+
+        /// <summary>
+        /// Render the Like button block frontend view.
+        /// </summary>
+        /// <param name="currentBlock">The current block instance.</param>
+        /// <returns>Result of the redirect to the current page.</returns>
+        public override ActionResult Index(LikeButtonBlock currentBlock)
+        {
+            var targetPage = pageRouteHelper.Page.ContentGuid.ToString();
+            var raterUser = this.User.Identity.GetUserId();
+            if (string.IsNullOrEmpty(raterUser)) raterUser = "anonymous-" + Guid.NewGuid();
+
+            // Create a rating block view model to fill the frontend block view
+            var blockModel = new LikeButtonBlockViewModel();
+
+            try
+            {
+                // Using the EPiServer Social Rating service, get the existing rating for the current user (rater) and page (target).
+                // Again, if there's no user logged into the site then we generate a random anonymous user, which should not currently 
+                // have any ratings, thus he/she will always be allow to rate the current page.
+                var ratingPage = ratingService.Get(
+                    new Criteria<RatingFilter>()
+                    {
+                        Filter = new RatingFilter()
+                        {
+                            Rater = Reference.Create(raterUser),
+                            Targets = new List<Reference> { Reference.Create(targetPage) }
+                        },
+                        PageInfo = new PageInfo() { PageSize = 1 }
+                    }
+                );
+
+                // Add the current Like rating, if any, to the block view model. If the user is logged 
+                // into the site and had previously liked the current page then the CurrentRating value
+                // should be 1 (LIKED_RATING).  Anonymous user Likes are generated with unique random users
+                // and thus the current anonymous user will never see a current rating value as he/she
+                // can Like the page indefinitely.
+                if (ratingPage.Results.Count() > 0)
+                {
+                    blockModel.CurrentRating = ratingPage.Results.ToList().FirstOrDefault().Value.Value;
+                }
```

Comment:
Maybe skip this check (and the lookup above) if the user is anonymous? That would prevent unnecessary pings to the service.